src/engines/algorithms_config.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List
+from enum import Enum
+from dataclasses import dataclass
+
+from algorithms import Algorithm
+from algorithms.clustering import KMeans, GMM
+from algorithms.associations import APriori
+from algorithms.classification import ExtraTrees
+from widgets.steps_widgets import AlgorithmStepsVisualization, APrioriStepsVisualization, KMeansStepsVisualization, \
+    ExtraTreesStepsVisualization, GMMStepsVisualization
+from widgets.results_widgets import KMeansResultsWidget, ExtraTreesResultsWidget, AlgorithmResultsWidget, \
+    APrioriResultsWidget, GMMResultsWidget
+from widgets.options_widgets import KMeansOptions, ExtraTreesOptions, AlgorithmOptions, AssociationRulesOptions, \
+    GMMOptions
+
+
+class AlgorithmTechniques(Enum):
+    CLUSTERING = "clustering"
+    ASSOCIATIONS = "associations"
+    CLASSIFICATION = "classification"
+
+    @classmethod
+    def list(cls) -> List[str]:
+        return list(map(lambda e: e.value, cls))
+
+
+@dataclass
+class AlgorithmClasses:
```

Comment:
not a fan of this name; maybe `AlgorithmConfig`?

Reply:
Same, but I didn't have idea how name this class.  `AlgorithmConfig` sounds good for me.